test/e2e/protocol/proposer.test.mjs

```diff
@@ -127,6 +127,20 @@ describe('Basic Proposer tests', () => {
     }
   });
 
+  it('should fail to register a proposer without a valid API key', async () => {
```

Comment:
How do you test that an end user without the right api key can’t call a private endpoint?

Reply:
You mean another test with no token at all? That makes sense, yes. I'll add it tomorrow

Reply:
Actually, if we want to be comprehensive, this test accounts for:

- You can't provide a wrong API key and access a private endpoint
- You must provide a correct API key to access a private endpoint

It should also account for:

- You can't access a private endpoint with an inexistent key
- You can access a public endpoint with, without, or with an invalid API key

Not sure if I'm covering all cases here, though

Reply:
Just submitted these missing tests